### PR TITLE
Watermark dom

### PR DESCRIPTION
**Pull request by shlroland**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/watermark-dom/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    watermark_color: string; // 水印字体颜色
+    watermark_fontsize: string; // 水印字体大小
+    watermark_alpha: number; // 水印透明度，要求设置在大于等于0.005
+    watermark_width: number; // 水印宽度
+    watermark_height: number; // 水印长度
+    watermark_angle: number; // 水印倾斜度数
+    watermark_parent_width: number; // 水印的总体宽度（默认值：body的scrollWidth和clientWidth的较大值）
+    watermark_parent_height: number; // 水印的总体高度（默认值：body的scrollHeight和clientHeight的较大值）
+    watermark_parent_node: Element | null; // 水印插件挂载的父元素element,不输入则默认挂在body上
+    monitor: boolean; // monitor 是否监控， true: 不可删除水印; false: 可删水印。
```
Comment (BendingBender) [inline types/watermark-dom/index.d.ts:34]: Please use doc comments:
```ts
    /**
     * 水印总体的id
     */
    watermark_id: string;
    ...
    /**
     * 水印的总体宽度
     * 
     * @default body的scrollWidth和clientWidth的较大值
     */
    watermark_parent_width: number;
    ...
```

File: types/watermark-dom/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+/*~ If this module is a UMD module that exposes a global variable 'myLib' when
+ *~ loaded outside a module loader environment, declare that global here.
+ *~ Otherwise, delete this declaration.
+ */
+export as namespace watermarkDom;
+export default watermarkDom;
+
+/*~ You can declare types that are available via importing the module */
+export interface Settings {
```
Comment (BendingBender) [inline types/watermark-dom/index.d.ts:14]: Why do you make the interface members non-optional and instead make the whole interface later in `init`/`load` `Partial`? It's not very ergonomic for your users and it increases compile times. You should instead make all of the interface members optional:
```ts
export interface Settings {
    watermark_id?: string;
    watermark_prefix?: string;
    watermark_txt?: string;
    watermark_x?: number;
    ...
}
```

**Review by BendingBender — CHANGES_REQUESTED** [review]

File: types/watermark-dom/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+     *~ or
+     *~   import * as yourMod from 'yourModule';
+     *~   yourMod.subProp.foo();
+     */
+    function init(settings: Partial<Settings>): void;
+
+    function load(settings: Partial<Settings>): void;
+
+    function remove(): void;
+}
```
Comment (BendingBender) [inline types/watermark-dom/index.d.ts:53]: The export is actually incorrect. The way it's specified you should be able to import this lib as `import { default } from 'watermark-dom';`. Which is clearly wrong, the library doesn't have a default export. This should instead be:
```suggestion
export function init(settings: Partial<Settings>): void;

export function load(settings: Partial<Settings>): void;

export function remove(): void;
```

File: types/watermark-dom/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+// Project: https://github.com/saucxs/watermark-dom (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
+// Definitions by: shenhaoliang <https://github.com/shlroland>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/*~ If this module is a UMD module that exposes a global variable 'myLib' when
+ *~ loaded outside a module loader environment, declare that global here.
+ *~ Otherwise, delete this declaration.
+ */
+export as namespace watermarkDom;
+export default watermarkDom;
```
Comment (BendingBender) [inline types/watermark-dom/index.d.ts:11]: This export is incorrect, there is no `default` export in the original lib.

**Review by BendingBender — CHANGES_REQUESTED** [review]

**Review by BendingBender — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]

**Review by andrewbranch — APPROVED** [review]

**Comment by shlroland** [comment]
> Ready to merge

Ready to merge

**Comment by shlroland** [comment]
Ready to merge

**Pull request merged**